test/reserve/DistributionHelper.test.ts

```diff
@@ -0,0 +1,325 @@
+  GoodReserveCDai,
+  DistributionHelperTestHelper,
+  DistributionBridgeMock,
+  ERC20
+} from "../../types";
+import { createDAO, increaseTime } from "../helpers";
+import { Contract } from "ethers";
+import { FormatTypes } from "@ethersproject/abi";
+import { recoverAddress } from "ethers/lib/utils";
+
+const BN = ethers.BigNumber;
+export const NULL_ADDRESS = ethers.constants.AddressZero;
+export const BLOCK_INTERVAL = 1;
+
+describe("DistributionHelper", () => {
+  let goodReserve: GoodReserveCDai;
+  let goodDollar: ERC20,
+    genericCall,
+    avatar,
+    founder,
+    signers,
+    setDAOAddress,
+    nameService,
+    cDai;
+
+  before(async () => {
+    [founder, ...signers] = await ethers.getSigners();
+
+    let {
+      controller: ctrl,
+      avatar: av,
+      gd,
+      identity,
+      setDAOAddress: sda,
+      setSchemes,
+      reserve,
+      cdaiAddress,
+      genericCall: gc,
+      nameService: ns
+    } = await createDAO();
+
+    nameService = ns;
+    genericCall = gc;
+    cDai = cdaiAddress;
+    avatar = av;
+    setDAOAddress = sda;
+
+    console.log("deployed dao", {
+      founder: founder.address,
+      gd,
+      identity,
+      avatar
+    });
+
+    goodDollar = (await ethers.getContractAt("IGoodDollar", gd)) as ERC20;
+
+    console.log("deployed contribution, deploying reserve...", {
+      founder: founder.address
+    });
+
+    goodReserve = reserve as GoodReserveCDai;
+  });
+
+  const fixture = async (wallets, provider) => {
+    const df = await ethers.getContractFactory("DistributionHelperTestHelper");
+    const rf = await ethers.getContractFactory("DistributionBridgeMock");
+
+    wallets = provider.getWallets();
+    const distHelper = (await waffle.deployContract(wallets[0], {
+      abi: JSON.parse(df.interface.format(FormatTypes.json) as string) as any[],
+      bytecode: df.bytecode
+    })) as DistributionHelperTestHelper;
+
+    await distHelper.initialize(nameService.address);
+
+    const bridge = (await waffle.deployContract(wallets[0], {
+      abi: JSON.parse(rf.interface.format(FormatTypes.json) as string) as any[],
+      bytecode: rf.bytecode
+    })) as DistributionBridgeMock;
+
+    await distHelper.setBridges(bridge.address, bridge.address);
+    return { distHelper, bridge };
+  };
+
+  it("should not allow to add recipient", async () => {
+    const { distHelper } = await waffle.loadFixture(fixture);
+
+    const recipient = signers[0];
+
+    await expect(
+      distHelper.addOrUpdateRecipient({
+        bps: 3000,
+        chainId: 1,
+        addr: recipient.address,
+        transferType: 0
+      })
+    ).to.be.revertedWith(
+      "is missing role 0x0000000000000000000000000000000000000000000000000000000000000000"
+    );
+  });
+
+  it("should allow to add recipient by avatar", async () => {
+    const { distHelper } = await waffle.loadFixture(fixture);
+
+    const recipient = signers[0];
+
+    const encodedCall = distHelper.interface.encodeFunctionData(
```

Comment:
I prefer runAsAvatarOnly, for both readability and the -not avatar should fail- check.
Same for the rest